src/components/menu/menu-directive.ts

```diff
@@ -65,6 +70,60 @@ export class MdMenu {
     this._showClickCatcher = bool;
   }
 
+  /**
+   * This method is used by the menu trigger to focus the first item when the menu is opened
+   * by the ENTER key.
+   * TODO: internal
+   */
+  _focusFirstItem() { this.items.first.focus(); }
+
+  // TODO(kara): update this when (keydown.downArrow) testability is fixed
+  private _checkKey(event: KeyboardEvent): void {
+    if (event.keyCode === DOWN_ARROW) {
+      this._focusNextItem();
+    } else if (event.keyCode === UP_ARROW) {
+      this._focusPreviousItem();
+    } else if (event.keyCode === TAB) {
+      this._checkTabDirection(event.shiftKey);
+    }
+  }
+
+  private _checkTabDirection(shiftPressed: boolean): void {
+    shiftPressed ? this._updateFocusedItem(-1) : this._updateFocusedItem(1);
+  }
+
+  private _emitCloseEvent(): void {
+    this._focusedItem = 0;
+    this.close.emit(null);
+  }
+
+  private _checkMenuBounds(): void {
+    if (this._focusedItem === this.items.length || this._focusedItem < 0) {
+      this._emitCloseEvent();
+    }
+  }
+
+  private _focusNextItem(): void {
+    this._updateFocusedItem(1);
+    this.items.toArray()[this._focusedItem].focus();
```

Comment:
It's weird that you call `updateFocusedItem` and then _also_ actually focus the item.